### ISSUE TITLE
First argument to mapArray should be recognised as a tracked scope

### ISSUE THREAD

**Issue by jfrere** [solidjs-community/eslint-plugin-solid#55]
**Describe the bug**

The first argument to `mapArray` is a function that is run within a tracked scope, but `eslint-plugin-solid` does not recognise it as a tracked scope.

**To Reproduce**

```typescript
export function createCustomStore(): Accessor<CustomItem[]> {
  const [store, updateStore] = createStore({
    /* ... */
  });

  return mapArray(
    // warning: This function should be passed to a tracked scope (like createEffect) or an event handler because it contains reactivity (solid/reactivity)
    () => store.path.to.field,
    (item) => ({
      /* ... */
    })
  );
}
```

**Expected behavior**

`mapArray` should be recognised as using tracked scopes.

**Environment (please complete the following information):**
- OS: Mac OS 13.1
- Node version (`node --version`): 19.2.0
- `eslint-plugin-solid` version (`npm list eslint-plugin-solid`/`yarn why eslint-plugin-solid`): 0.9.1
- `eslint` version (`npm list eslint`/`yarn why eslint`): 8.29.0

**Additional context**

I've split this problem off from #52 as discussed there
